src/lighteval/metrics/metrics_corpus.py

```diff
@@ -103,7 +103,11 @@ def __init__(self, metric_type: str):
     def compute(self, items: list[GenerativeCorpusMetricInput]) -> float:
         """Computes the metric score over all the corpus generated items, by using the sacrebleu implementation."""
         golds = [i.golds for i in items]
-        preds = [as_list(i.preds) for i in items]
+        preds = []
+        for i in items:
+            if len(i.preds) > 1:
+                hlog_warn("Multiple predictions present, keeping only the first prediction.")
+            preds.append(i.preds[0])
```

Comment:
```suggestion
       for i in items:
            pred = as_list(i.preds)
            if len(pred) > 1:
                hlog_warn("Multiple predictions present, keeping only the first prediction.")
            preds.append(pred[0])
```

Reply:
(we need to cast i.preds to a list in case it's something else)

Reply:
Done